packages/@aws-cdk/aws-vpcv2-alpha/lib/vpc-v2-base.ts

```diff
@@ -0,0 +1,368 @@
+import { Resource, Annotations } from 'aws-cdk-lib';
+import { IVpc, ISubnet, SubnetSelection, SelectedSubnets, EnableVpnGatewayOptions, VpnGateway, VpnConnectionType, CfnVPCGatewayAttachment, CfnVPNGatewayRoutePropagation, VpnConnectionOptions, VpnConnection, ClientVpnEndpointOptions, ClientVpnEndpoint, InterfaceVpcEndpointOptions, InterfaceVpcEndpoint, GatewayVpcEndpointOptions, GatewayVpcEndpoint, FlowLogOptions, FlowLog, FlowLogResourceType, SubnetType, SubnetFilter, CfnVPCCidrBlock } from 'aws-cdk-lib/aws-ec2';
+// eslint-disable-next-line no-duplicate-imports
+import { allRouteTableIds, flatten, subnetGroupNameFromConstructId } from '../lib/util';
+import { IDependable, Dependable, IConstruct } from 'constructs';
+
+/**
+ * Placeholder to see what extra props we might need,
+ * will be added to original IVPC
+ */
+export interface IVpcV2 extends IVpc {
+  /**
+   * The secondary CIDR blocks associated with the VPC.
+   *
+   * For more information, see the {@link https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.Vpc.html#configuring-secondary-cidr-blocks}.
+   */
+  readonly secondaryCidrBlock: CfnVPCCidrBlock[];
+
+  /**
+   * The primary IPv4 CIDR block associated with the VPC.
+   *
+   * For more information, see the {@link https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.Vpc.html#configuring-ipv4-cidr-block}.
+   */
+  readonly ipv4CidrBlock: string;
+}
+
+/**
+ * Base class for creating a VPC (Virtual Private Cloud) in AWS.
+ *
+ * For more information, see the {@link https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.Vpc.html|AWS CDK Documentation on VPCs}.
+ */
+export abstract class VpcV2Base extends Resource implements IVpcV2 {
+
+  /**
+     * Identifier for this VPC
+     */
+  public abstract readonly vpcId: string;
+
+  /**
+     * Arn of this VPC
+     */
+  public abstract readonly vpcArn: string;
+
+  /**
+     * CIDR range for this VPC
+     */
+  public abstract readonly vpcCidrBlock: string;
+
+  /**
+     * List of public subnets in this VPC
+     */
+  public readonly publicSubnets: ISubnet[] = [];
```

Comment:
since we 're not creating any defaults here in the new construct ,it is empty but it will populated with some defaults in future to provide a similar experience, also these are needed for selection subnets but it is currently under user control to a set a subnet as 'Private' or 'public'

Reply:
I think we should remove the assignment here to an empty array. You already moved it to the constructor

Reply:
its not initialized in vpc-base constructor class, hence needed